x/stake/client/rest/query.go

```diff
@@ -16,33 +19,97 @@ import (
 const storeName = "stake"
 
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *wire.Codec) {
+
+	// GET /stake/delegators/{delegatorAddr} // Get all delegations (delegation, undelegation and redelegation) from a delegator
 	r.HandleFunc(
-		"/stake/{delegator}/delegation/{validator}",
-		delegationHandlerFn(cliCtx, cdc),
+		"/stake/delegators/{delegatorAddr}",
+		delegatorHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+
+	// GET /stake/delegators/{delegatorAddr}/txs // Get all staking txs (i.e msgs) from a delegator
+	r.HandleFunc(
+		"/stake/delegators/{delegatorAddr}/txs",
+		delegatorTxsHandlerFn(cliCtx, cdc),
 	).Methods("GET")
 
+	// // GET /stake/delegators/{delegatorAddr}/validators // Query all validators that a delegator is bonded to
+	// r.HandleFunc(
+	// 	"/stake/delegators/{delegatorAddr}/validators",
+	// 	delegatorValidatorsHandlerFn(cliCtx, cdc),
+	// ).Methods("GET")
+
+	// GET /stake/delegators/{delegatorAddr}/validators/{validatorAddr} // Query a validator that a delegator is bonded to
+	// r.HandleFunc(
+	// 	"/stake/delegators/{delegatorAddr}/validators",
+	// 	delegatorValidatorHandlerFn(cliCtx, cdc),
+	// ).Methods("GET")
```

Comment:
delete all commented out code in this file?

Reply:
@rigelrozanski This is a `TODO`. Should we delete it or leave it as commented ? We already have the functionality but I spoke with @faboweb and agreed to leave it for later...

Reply:
I'll make an issue for that and delete the comments

Reply:
See #1939